internal/database/postgres.go

```diff
@@ -339,6 +341,13 @@ WHERE Received > $1 AND Received <= (SELECT MAX(SampleEndTime) FROM TempDownsamp
 		return err
 	}
 
+	_, err = tx.Exec(delete_query)
+
+	if err != nil {
+		tx.Rollback()
```

Comment:
The error this rollback returns should be combined with the exec error using `errors.Join()`